runtime/testing/jvm/src/aws/smithy/kotlin/runtime/testing/ManualDispatchTestBase.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+package aws.smithy.kotlin.runtime.testing
+
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineScope
+import kotlinx.coroutines.test.runBlockingTest
+import kotlin.test.assertEquals
+import kotlin.test.assertNotEquals
+
+// modeled after https://github.com/ktorio/ktor/blob/78e36790cdbb30313dfbd23b174bffe805d26dca/ktor-io/common/test/io/ktor/utils/io/ByteChannelTestBase.kt
+// but implemented using [kotlinx-coroutines-test](https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-test)
+// rather than rolling our own dummy coroutines dispatcher
+/**
+ * Test suspend functions with precise control over coordination. This allows testing that suspension happens at the
+ * expected point in time using the provided [expect] function, e.g.:
```

Comment:
**Question**: Why a language-level comment and a KDoc? Why not just include the provenance in the KDoc itself?

Reply:
Because it's unimportant to how it's used but it is of interest to someone that needs to maintain it. Thats the line I draw at least